homeassistant/components/sma/config_flow.py

```diff
@@ -0,0 +1,141 @@
+        """Initialize."""
+        self._data = {
+            CONF_HOST: vol.UNDEFINED,
+            CONF_SSL: False,
+            CONF_VERIFY_SSL: True,
+            CONF_GROUP: GROUPS[0],
+            CONF_PASSWORD: vol.UNDEFINED,
+            CONF_SENSORS: [],
+            CONF_CUSTOM: {},
+            DEVICE_INFO: {},
+        }
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> dict[str, Any]:
+        """First step in config flow."""
+        errors = {}
+        if user_input is not None:
+            self._data[CONF_HOST] = user_input[CONF_HOST]
+            self._data[CONF_SSL] = user_input[CONF_SSL]
+            self._data[CONF_VERIFY_SSL] = user_input[CONF_VERIFY_SSL]
+            self._data[CONF_GROUP] = user_input[CONF_GROUP]
+            self._data[CONF_PASSWORD] = user_input[CONF_PASSWORD]
+
+            try:
+                self._data[DEVICE_INFO] = await validate_input(self.hass, user_input)
+            except aiohttp.ClientError:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except CannotRetrieveDeviceInfo:
+                errors["base"] = "cannot_retrieve_device_info"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if not errors:
+                await self.async_set_unique_id(self._data[DEVICE_INFO]["serial"])
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=self._data[CONF_HOST], data=self._data
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=self._data[CONF_HOST]): cv.string,
+                    vol.Optional(CONF_SSL, default=self._data[CONF_SSL]): cv.boolean,
+                    vol.Optional(
+                        CONF_VERIFY_SSL, default=self._data[CONF_VERIFY_SSL]
+                    ): cv.boolean,
+                    vol.Optional(CONF_GROUP, default=self._data[CONF_GROUP]): vol.In(
+                        GROUPS
+                    ),
+                    vol.Required(CONF_PASSWORD): cv.string,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(
+        self, import_config: dict[str, Any] | None
+    ) -> dict[str, Any]:
+        """Import a config flow from configuration."""
+        device_info = await validate_input(self.hass, import_config)
+        import_config[DEVICE_INFO] = device_info
+
+        # If unique is configured import was already run
+        # This means remap was already done, so we can abort
+        await self.async_set_unique_id(device_info["serial"])
+        self._abort_if_unique_id_configured(import_config)
```

Comment:
We normally don't support updating a config entry with configuration.yaml data. It should be a one time import unless there are special reasons otherwise.

Reply:
I've kept this in to support the legacy way of defining custom sensors and allow easy testing of new sensors.
https://github.com/rklomp/home-assistant.io/blob/next/source/_integrations/sma.markdown#custom-sensors

Reply:
Maybe we can add an options flow for adding a custom sensor config?